api/hasura/actions/upload_profile_avatar.ts

```diff
@@ -0,0 +1,107 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { v4 as uuidv4 } from 'uuid';
+
+import { gql } from '../../../api-lib/Gql';
+import { resizeAvatar } from '../../../api-lib/images';
+import { deleteImage, uploadImage } from '../../../api-lib/s3';
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  updateProfileAvatarInput,
+} from '../../../src/lib/zod';
+
+// this buffer is because the base64 size of the file is approximately 33% bigger than the underlying file
+// and the client limits file to 10MB - we need to make sure we have enough room to handle the base64 overhead
+const base64Buffer = 1.4;
+const MAX_IMAGE_BYTES_LENGTH = 10 * 1024 * 1024 * base64Buffer; // 10MB+buffer
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  // parse the input
+  const {
+    input: { object: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    updateProfileAvatarInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+
+  try {
+    // base64 decode the provided image data
+    const imageBytes = Buffer.from(input.image_data_base64, 'base64');
+
+    // file size check
+    if (imageBytes.byteLength > MAX_IMAGE_BYTES_LENGTH) {
+      return res.status(400).json({
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413 might be a nice code to more accurately descripe the error here.

Reply:
I switched this to 413 and moved it to images.ts